contracts/testHelpers/NoLimitSubdomains.sol

```diff
@@ -0,0 +1,59 @@
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity >=0.5.3;
+pragma experimental ABIEncoderV2;
+
+import "./../ColonyStorage.sol";
+
+contract NoLimitSubdomains is ColonyStorage {
+  function addDomain(uint256 _parentDomainId) public
+  stoppable
+  auth
+  domainExists(_parentDomainId)
+  {
+    uint256 parentSkillId = domains[_parentDomainId].skillId;
+
+    // Setup new local skill
+    IColonyNetwork colonyNetwork = IColonyNetwork(colonyNetworkAddress);
+    uint256 newLocalSkill = colonyNetwork.addSkill(parentSkillId);
+
+    // Add domain to local mapping
+    initialiseDomain(newLocalSkill);
+  }
+
+  // We copy this function from Colony.sol to here because it's private, not internal, so we can't access it even if we inherit
+  function initialiseDomain(uint256 _skillId) private skillExists(_skillId) {
```

Comment:
Can we make `initialiseDomain` internal in `Colony` then?

Reply:
I generally don't like changing the codebase to make tests easier to write, but given you're both in favour of it...

Reply:
I agree. But I disagree this is an example of said anti-working pattern.
We are extending the inheritance model here and we are reviewing the functions that trickle through and those that don't and updating accordingly. I can't imagine we could have predicted we'll be fudging domain creation.
Our use of `private` functions has been a bit more liberal, in that we prefer them even for functions only called from within the current contract even if they are potentially useful downstream in future contracts.
I personally prefer this more conservative approach but I'm happy to re-review our `internal`/`private` modifiers use in line with what I think are your expectations.